src/poller/poller-core/index.js

```diff
@@ -367,67 +406,110 @@ async function parseAndEnrichPayload(payload) {
   const clusters = await configValidator.parseAndAssertValidConfig(payload);
   const clustersFound = [];
 
-  for (let idx = 0; idx < clusters.length; idx++) {
+  for (let clusterIdx = 0; clusterIdx < clusters.length; clusterIdx++) {
+    const customMetrics =
+      /** @type {MemorystoreClusterMetric[]} */
+      (clusters[clusterIdx].metrics);
     // Merge in the defaults
-    clusters[idx] = {...baseDefaults, ...clusters[idx]};
+    clusters[clusterIdx] = {...baseDefaults, ...clusters[clusterIdx]};
```

Comment:
nit: 
This block would be a whole lot more readable if you did: 
```
clusters[clusterIdx] = {...baseDefaults, ...clusters[clusterIdx]};
const cluster = clusters[clusterIdx];
```

Reply:
Fair -- done.